### PR TITLE
Dashboard: handle the supported types of environments

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?
This PR adds to Dashboard the handling of the supported types of recipes. The end user is allowed to use only supported recipe types to build a new stack. 
Also, a workspace based on unsupported recipe type is prevented to start from navbar and Workspaces List page.

screenshot-localhost-3000-2018-01-10-12-31-25-185
selection_059
*Navbar:* prevent to start an unsupported workspace
selection_041
*Workspaces List:* prevent to start a workspace and mark it as unsupported 
selection_042

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/7811

### CODE REVIEW

**Comment by akurinnoy** [comment]
ci-test

**Comment by akurinnoy** [comment]
@ashumilova @benoitf @olexii4 
please review

**Comment by ghost** [comment]
@slemeur I'll cover this in new docs

File: dashboard/src/app/workspaces/create-workspace/create-workspace.controller.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ import {CreateWorkspaceSvc} from './create-workspace.service';
 import {NamespaceSelectorSvc} from './namespace-selector/namespace-selector.service';
 import {StackSelectorSvc} from './stack-selector/stack-selector.service';
 import {RandomSvc} from '../../../components/utils/random.service';
+import {CheNotification} from "../../../components/notification/che-notification.factory";
```
Comment (ashumilova) [inline dashboard/src/app/workspaces/create-workspace/create-workspace.controller.ts]: " -> '

File: dashboard/src/app/workspaces/workspace-details/environments/environments.controller.ts [hunk 2]
```diff
@@ -111,6 +135,15 @@ export class WorkspaceEnvironmentsController {
     }
   }
 
+  /**
+   * Returns true if the recipe type is scalable.
+   *
+   * @returns {boolean}
+   */
+  isScalable(): boolean {
```
Comment (ashumilova) [inline dashboard/src/app/workspaces/workspace-details/environments/environments.controller.ts]: Propose the term - isMultiContainer/isMultiMachine ?

**Review by ashumilova — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Review by olexii4 — APPROVED** [review]

File: dashboard/src/app/workspaces/create-workspace/create-workspace.controller.ts (hunk 1 above)
Comment (akurinnoy) [reply dashboard/src/app/workspaces/create-workspace/create-workspace.controller.ts]: done

File: dashboard/src/app/workspaces/workspace-details/environments/environments.controller.ts (hunk 2 above)
Comment (akurinnoy) [reply dashboard/src/app/workspaces/workspace-details/environments/environments.controller.ts]: done

**Pull request merged**